nrfutil/device/program.ts

```diff
@@ -62,33 +62,41 @@ export const isNordicDfuProgrammingOptions = (
     !isMcuBootProgrammingOptions(options) &&
     (options as NordicDfuProgrammingOptions).mcuEndState !== undefined;
 
-export const programmingOptionsToArgs = (options?: ProgrammingOptions) => {
+const programmingOptionsToStrings = (options?: ProgrammingOptions) => {
     if (!options) return [];
 
-    const args: string[] = [];
-
     if (isJLinkProgrammingOptions(options)) {
-        if (options.chipEraseMode)
-            args.push(`chip_erase_mode=${options.chipEraseMode}`);
-        if (options.reset) args.push(`reset=${options.reset}`);
-        if (options.verify) args.push(`verify=${options.verify}`);
-    } else if (isMcuBootProgrammingOptions(options)) {
-        if (options.mcuEndState)
-            args.push(`mcu_end_state=${options.mcuEndState}`);
-        if (options.netCoreUploadDelay)
-            args.push(
+        return [
+            `chip_erase_mode=${options.chipEraseMode}`,
+            options.reset && `reset=${options.reset}`,
+            options.verify && `verify=${options.verify}`,
+        ].filter(Boolean);
+    }
+
+    if (isMcuBootProgrammingOptions(options)) {
+        return [
+            options.mcuEndState && `mcu_end_state=${options.mcuEndState}`,
+            options.target && `target=${options.target}`,
+            options.netCoreUploadDelay &&
                 `net_core_upload_delay=${Math.round(
                     options.netCoreUploadDelay
-                )}`
-            );
-        if (options.target) args.push(`target=${options.target}`);
-    } else if (isNordicDfuProgrammingOptions(options)) {
-        if (options.mcuEndState)
```

Comment:
Why not keep this check?

Reply:
Because there are only exactly three possibilities for what type the options can be. So it made the code simpler to remove it.

An alternative would have been to throw an exception in this case. But I saw no need for that and so decided to use the “less code” alternative.